clients/deltalake/core/src/test/java/org/projectnessie/deltalake/ITDeltaLog.java

```diff
@@ -51,6 +57,91 @@ protected static void createDelta() {
         .set("spark.sql.catalog.spark_catalog", "org.apache.spark.sql.delta.catalog.DeltaCatalog");
   }
 
+  @Test
+  // Delta < 0.8 w/ Spark 2.x doesn't support multiple branches well (warnings when changing the configuration)
+  @DisabledIfSystemProperty(named = "skip-multi-branch-tests", matches = "true")
+  void testMultipleBranches() throws Exception {
+    String csvSalaries1 = ITDeltaLog.class.getResource("/salaries1.csv").getPath();
+    String csvSalaries2 = ITDeltaLog.class.getResource("/salaries2.csv").getPath();
+    String csvSalaries3 = ITDeltaLog.class.getResource("/salaries3.csv").getPath();
+    String pathSalaries = new File(tempPath, "salaries").getAbsolutePath();
+
+    spark.sql(String.format("CREATE TABLE IF NOT EXISTS test_multiple_branches (Season STRING, Team STRING, Salary STRING, "
+        + "Player STRING) USING delta LOCATION '%s'", pathSalaries));
+    Dataset<Row> salariesDf1 = spark.read().option("header", true).csv(csvSalaries1);
+    salariesDf1.write().format("delta").mode("overwrite").save(pathSalaries);
+
+    Dataset<Row> count1 = spark.sql("SELECT COUNT(*) FROM test_multiple_branches");
+    Assertions.assertEquals(15L, count1.collectAsList().get(0).getLong(0));
+
+    Reference mainBranch = nessieClient.getTreeApi().getReferenceByName("main");
+
+    Reference devBranch = nessieClient.getTreeApi().createReference(Branch.of("testMultipleBranches", mainBranch.getHash()));
+
+    spark.sparkContext().getConf().set("spark.hadoop." + CONF_NESSIE_REF, devBranch.getName());
+    spark.sparkContext().hadoopConfiguration().set(CONF_NESSIE_REF, devBranch.getName());
+
+    Dataset<Row> salariesDf2 = spark.read().option("header", true).csv(csvSalaries2);
+    salariesDf2.write().format("delta").mode("append").save(pathSalaries);
+
+    Dataset<Row> count2 = spark.sql("SELECT COUNT(*) FROM test_multiple_branches");
+    Assertions.assertEquals(30L, count2.collectAsList().get(0).getLong(0));
+
+    spark.sparkContext().getConf().set("spark.hadoop.nessie.ref", "main");
+    spark.sparkContext().hadoopConfiguration().set("nessie.ref", "main");
+
+    Dataset<Row> salariesDf3 = spark.read().option("header", true).csv(csvSalaries3);
+    salariesDf3.write().format("delta").mode("append").save(pathSalaries);
+
+    Dataset<Row> count3 = spark.sql("SELECT COUNT(*) FROM test_multiple_branches");
+    Assertions.assertEquals(35L, count3.collectAsList().get(0).getLong(0));
+  }
+
+  @Test
+  // Delta < 0.8 w/ Spark 2.x doesn't support multiple branches well (warnings when changing the configuration)
+  @DisabledIfSystemProperty(named = "skip-multi-branch-tests", matches = "true")
+  void testCommitRetry() throws Exception {
+    String csvSalaries1 = ITDeltaLog.class.getResource("/salaries1.csv").getPath();
```

Comment:
Any reason to use csvs when everywhere else in spark tests we create simple few row in memory datasets?

Reply:
A mixture of "laziness and reproduction of the original issue". I wanted this test to be close to what the demo in `nessie-demos` looks like.